pkg/tfvars/openstack/openstack.go

```diff
@@ -56,16 +57,21 @@ func TFVars(masterConfigs []*v1alpha1.OpenstackProviderSpec, cloud string, exter
 	}
 
 	cfg := &config{
-		ExternalNetwork:         externalNetwork,
-		Cloud:                   cloud,
-		FlavorName:              masterConfigs[0].Flavor,
-		APIFloatingIP:           apiFloatingIP,
-		IngressFloatingIP:       ingressFloatingIP,
-		APIVIP:                  apiVIP,
-		IngressVIP:              ingressVIP,
-		ExternalDNS:             externalDNS,
-		MachinesSubnet:          machinesSubnet,
-		MasterAvailabilityZones: zones,
+		ExternalNetwork:                   externalNetwork,
+		Cloud:                             cloud,
+		FlavorName:                        masterConfigs[0].Flavor,
+		APIFloatingIP:                     apiFloatingIP,
+		IngressFloatingIP:                 ingressFloatingIP,
+		APIVIP:                            apiVIP,
+		IngressVIP:                        ingressVIP,
+		ExternalDNS:                       externalDNS,
+		MachinesSubnet:                    machinesSubnet,
+		MasterAvailabilityZones:           zones,
+		MasterRootVolumesAvalabilityZones: []string{""},
+	}
+
+	if mpool != nil && mpool.RootVolume != nil && mpool.RootVolume.Zones != nil {
```

Comment:
Agreed, I was also confused by this `mpool.RootVolume.Zones != nil` check at first and thought it could cause a bug. Looks like it's fine and we respect the user provided mpool after all, but this code it still confusing and not necessary.